dothething/dothething/ClipsView.swift

```diff
@@ -35,71 +34,101 @@ struct ClipsView: View {
-                .opacity(clipsViewModel.shareDisabled ? 0.5 : 1)
-            }
-
-            if !clipsViewModel.errorText.isEmpty && clipsViewModel.clips.isEmpty {
-                Text(clipsViewModel.errorText)
-                    .padding(.top)
-                    .foregroundColor(Color(UIColor.systemGray))
-            } else if clipsViewModel.isLoading {
-                ProgressView()
-                    .progressViewStyle(.circular)
-                    .padding(.top)
-            }
-            
-            ScrollView {
-                LazyVGrid(columns: [GridItem(.adaptive(minimum: 108, maximum: 108), spacing: 16)]) {
-                    ForEach(clipsViewModel.clips, id: \.self) { clip in
-                        ClipView(clip: clip)
+                .padding(.top, -12)
+                .padding(.bottom, -12)
+                
+                ZStack {
+                     Rectangle()
+                        .frame(height: 48)
+                        .foregroundColor(.accentColor)
+
+                    Text("code: \(clipsViewModel.code)")
+                        .font(.custom("Montserrat-Light", size: 24))
+                        .foregroundColor(.accentColor)
+                        .colorInvert()
+                        .onTapGesture {
+                            print("Code tapped")
+                            UIPasteboard.general.string = clipsViewModel.code
+                        }
+                }
+                
+                Button(action: {
+                    if clipsViewModel.buttonText == "place a domino" {
+                        clipsViewModel.uploadButtonPressed()
+                    } else if clipsViewModel.buttonText == "share this rally" {
+                        clipsViewModel.shareButtonPressed()
                     }
+                }) {
+                    Text(clipsViewModel.buttonText)
+                        .font(.custom("Montserrat-Light", size: 24, relativeTo: .title))
+                        .foregroundColor(.accentColor)
+                        .colorInvert()
+                        .padding()
+                        .frame(width: 300, height: 50)
+                        .background(Color.accentColor)
+                        .cornerRadius(50)
                 }
-                .padding(.top, 4)
+                .opacity(clipsViewModel.uploadDisabled ? 0.5 : 1)
+                .padding(.top)
+                .padding(.bottom)
             }
-            
-            Text("Code: \(clipsViewModel.code)")
-                .padding()
-                .onTapGesture {
-                    print("Code tapped")
-                    UIPasteboard.general.string = clipsViewModel.code
+            .onAppear {
+                if code.isEmpty {
+                    print("Entered ClipsView with empty code")
+                    clipsViewModel.openImagePicker()
+                } else {
+                    clipsViewModel.enterCode(code: code)
                 }
```

Comment:
The changes in this section add a new onAppear handler that checks the code and either opens the image picker or enters the code, depending on whether the code is empty or not. This is a good improvement, as it ensures the app handles both scenarios seamlessly.